src/ui/src/core/navigation.ts

```diff
@@ -60,29 +51,23 @@ export function serializeParsedHash(): SerializedHash {
 	return { pageKey: parsedHash.pageKey, routeVars };
 }
 
-function setHash(parsedHash: ParsedHash) {
+export function serializeParsedHashAsString(parsedHash: ParsedHash) {
 	const { pageKey, routeVars } = parsedHash;
 
 	let hash = "";
 	if (pageKey) {
 		hash += `${encodeURIComponent(pageKey)}`;
 	}
 
-	if (routeVars.keys.length > 0) {
```

Comment:
This was a bug, since we need to use `.size`

```js
> const map = new Map()
undefined
> map.keys.length
0
> map.set('a','b')
Map(1) { 'a' => 'b' }
> map.keys.length
0
> map.size
1
```